pkg/sql/logictest/testdata/logic_test/privilege_builtins

```diff
@@ -1056,3 +1056,53 @@ query B
 SELECT has_column_privilege('hcp_test'::REGCLASS, 4, 'SELECT')
 ----
 true
+
+# Regression test for #58254.
```

Comment:
nit: could the comment say that it tests privilege inheritance for roles? (you can keep the issue number too, the idea is just to save people a copy+paste)

Reply:
Done!